src/mlpack/core/data/load_impl.hpp

```diff
@@ -63,17 +63,24 @@ void TransposeTokens(std::vector<std::vector<std::string>> const &input,
 
 } // namespace details
 
-template<typename eT>
-bool inline inplace_transpose(arma::Mat<eT>& X)
+template <typename MatType>
+bool inline inplace_transpose(MatType& X, bool fatal)
 {
   try
   {
     X = arma::trans(X);
-    return false;
+    return true;
   }
-  catch (std::bad_alloc&)
+  catch (const std::exception& e)
   {
-    return true;
+    if (fatal)
+      Log::Fatal << "\nTranspose Operation Failed.\n"
+          "Exception: " << e.what() << std::endl;
+    else
+      Log::Warn << "\nTranspose Operation Failed.\n"
+          "Exception: " << e.what() << std::endl;
+    
+    return false;
```

Comment:
Thanks!  This at least does something better than before.  Maybe later we should revisit it... it doesn't look like it actually attempts to do the transpose in-place (maybe Armadillo does that internally now, I'm not sure).  So in my view the function is a candidate for removal or overhaul, but, we can leave that for another time. :)